java/mapreduce/src/test/org/apache/orc/mapred/TestOrcStruct.java

```diff
@@ -123,4 +123,18 @@ public void testCompare() {
     assertEquals(1 , left.compareTo(right));
     assertEquals(-1, right.compareTo(left));
   }
+
+  @Test
+  public void testSchemaInCompare() {
+    TypeDescription leftType = TypeDescription.fromString("struct<s:string,i:int>");
+    TypeDescription rightType = TypeDescription.fromString("struct<s:string,j:bigint>");
+    OrcStruct left = new OrcStruct(leftType);
+    OrcStruct right = new OrcStruct(rightType);
+    assertEquals(-1, left.compareTo(right));
+    assertEquals(1, right.compareTo(left));
+    left.setAllFields(new Text("123"), new IntWritable(123));
```

Comment:
I checked out your branch and got a compilation error for setAllFields, which was introduced for ORC-52.